schema/home/home_page_artwork_modules.js

```diff
@@ -1,79 +1,169 @@
+    },
+    FOLLOWED_GENES: {
+      value: 'genes',
+    },
+    GENERIC_GENES: {
+      value: 'generic_gene',
+    },
+  },
+});
+
 const HomePageArtworkModules = {
   type: new GraphQLList(HomePageArtworkModuleType),
   description: 'Artwork modules to show on the home screen',
   args: {
     max_rails: {
       type: GraphQLInt,
-      description: 'Maximum number of modules to return',
+      description: 'Maximum number of modules to return, disable limit with a negative number',
       defaultValue: 8,
     },
+    max_followed_gene_rails: {
+      type: GraphQLInt,
+      description: 'Maximum number of followed genes to return, disable with a negative number',
+      defaultValue: 1,
+    },
+    order: {
+      type: new GraphQLList(HomePageArtworkModuleTypes),
+      description: 'The preferred order of modules, defaults to order returned by Gravity',
+    },
   },
-  resolve: (root, { max_rails }, request, { rootValue: { accessToken, userID } }) => {
+  resolve: (
+    root,
+    { max_rails, max_followed_gene_rails, order },
+    request,
+    { rootValue: { accessToken, userID } }
+  ) => {
     // If user is logged in, get their specific modules
     if (accessToken) {
       return gravity.with(accessToken)('me/modules').then((response) => {
-        const modules = map(keys(response), (key) => {
-          const display = response[key];
-          return { key, display };
-        });
-        const filteredModules = filterModules(modules, max_rails);
+        const modulesToDisplay = map(keys(response), (key) => ({ key, display: response[key] }));
+        return addFollowedGenes(accessToken, modulesToDisplay, max_followed_gene_rails)
+          .then(allModulesToDisplay => {
+            let modules = allModulesToDisplay;
+
+            modules = filterModules(modules, max_rails);
+            modules = reorderModules(modules, order);
 
-        // For the related artists rail, we need to fetch a random
-        // set of followed artist + related artist initially
-        // and pass it along so that any placeholder titles are consistent
-        const relatedArtistIndex = findIndex(filteredModules, { key: 'related_artists' });
+            // For the related artists rail, we need to fetch a random
+            // set of followed artist + related artist initially
+            // and pass it along so that any placeholder titles are consistent
+            const relatedArtistIndex = findIndex(modules, { key: 'related_artists' });
 
-        if (relatedArtistIndex > -1) {
-          return Promise.resolve(relatedArtist(accessToken, userID))
-          .then((artistPair) => {
-            if (artistPair) {
-              const { artist, sim_artist } = artistPair;
+            if (relatedArtistIndex > -1) {
+              return Promise.resolve(relatedArtist(accessToken, userID))
+                .then((artistPair) => {
+                  if (artistPair) {
+                    const { artist, sim_artist } = artistPair;
```

Comment:
Weird, I deleted that comment, but it's semi valid, everything in that file is in snake case FWIW